### PR TITLE
Added pin removal recipe

### PR DESCRIPTION
**Pull request by Razharas**
Added pin removal recipe
You need plasmacutter to do it and some other shit

### CODE REVIEW

**Comment by GunHog** [comment]
Removing the pin should destroy it, else you will have to make it not possible to farm pins from other, less useful gun items.

**Comment by Killbasa** [comment]
What is the point of this? When would you need to remove/destroy a pin? The only thing i see is using the practice laser to actually kill people.

**Comment by phil235** [comment]
@Killbasa The practice laser isn't a real laser, no matter what pin is inside it.

**Comment by Killbasa** [comment]
I just checked it and yeah i seem to have forgot about that. I think it just proves my point even more that it isn't really a useful thing to add.

**Comment by vista-** [comment]
Usage scenario: R&D gets stronk and supports fasc- I mean, security, all security guns are brought in to be upgraded to loyalty firing pins.

**Comment by GunHog** [comment]
I am more worried about people farming electronic pins from utility/practice guns and using them in real guns. But that concern is now addressed. Having Sec bring in their guns for loyalty pin upgrading is a good thing in my book, as Security is supposed to work with Science to improve their gear.

**Comment by Killbasa** [comment]
Loyality pins are a bad idea because it's a straight buff to sec and nerf to antags. Let's say an antag takes down an officer, he can't use the taser/energy/laser gun they might have. I think the way things are now is fine.

**Comment by Cheridan** [comment]
Pins were always intended to be swappable, I just wanted it to be something more difficult than attack_self, hit with screwdriver, etc. This works. Now I can cheat at laser tag by putting standard pins in tag guns :^)

I believe there's already code for swapping out certain pins by hitting the gun with them? You might want to check that out and remove it if its obsolete.

**Comment by Razharas** [comment]
I never heard about pin switching, i heard a lot of complains about there arent being such a feature

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -54,6 +54,14 @@
 	if(pin)
 		pin = new pin(src)
 
+/obj/item/weapon/gun/CheckParts()
+	var/obj/item/weapon/gun/G = locate(/obj/item/weapon/gun) in contents
+	if(G)
+		G.loc = loc
+		qdel(G.pin)
+		G.pin = null
+		visible_message("[G] can now fit a new pin, but old one was destroyed in the process.")
```
Comment (RemieRichards) [inline code/modules/projectiles/gun.dm]: "but THE old one..." would me more grammatically correct.

**Comment by Cheridan** [comment]
#8595 added the ability to remove test-range pins with other pins. I think that's ok to be left in as a usability thing.

**Comment by Razharas** [comment]
I dont understand hard englisg
Deal or no deal?

**Pull request merged**